bee-protocol/src/worker/solidifier/transaction.rs

```diff
@@ -35,15 +41,16 @@ impl TransactionSolidifierWorker {
 
     // TODO is the index even needed ? We request one milestone at a time ? No PriorityQueue ?
 
-    async fn solidify(&self, hash: Hash, index: u32) -> bool {
+    async fn solidify(&self, hash: Hash, index: MilestoneIndex) -> bool {
         let mut missing_hashes = HashSet::new();
 
-        tangle().walk_approvees_depth_first(
+        traversal::walk_approvees_dfs(
+            &tangle().inner,
             hash,
-            |_| {},
-            |vertex| !vertex.is_solid(),
-            |missing_hash| {
-                missing_hashes.insert(*missing_hash);
+            |_, v| v.get_meta().contains(Flags::SOLID),
```

Comment:
Do we want to have methods for all the flags ? Feels weird to have to expose the actual flag.

Reply:
Sure! I like this idea. Have provided reasonable methods.